src/main.rs

```diff
@@ -464,29 +464,33 @@ fn main_try() -> Result<()> {
         }
     }
 
-    if config.gdb.enabled && config.rtt.enabled {
-        return Err(anyhow!(
-            "Unfortunately, at the moment, only GDB OR RTT are possible."
-        ));
-    }
+    // if config.gdb.enabled && config.rtt.enabled {
+    //     return Err(anyhow!(
+    //         "Unfortunately, at the moment, only GDB OR RTT are possible."
+    //     ));
+    // }
+
+    let session = Arc::new(Mutex::new(session));
 
     if config.gdb.enabled {
-        let gdb_connection_string = config
-            .gdb
-            .gdb_connection_string
-            .as_deref()
-            .or_else(|| Some("127.0.0.1:1337"));
-        // This next unwrap will always resolve as the connection string is always Some(T).
-        logging::println(format!(
-            "Firing up GDB stub at {}.",
-            gdb_connection_string.as_ref().unwrap(),
-        ));
-        if let Err(e) = probe_rs_gdb_server::run(gdb_connection_string, session) {
-            logging::eprintln("During the execution of GDB an error was encountered:");
-            logging::eprintln(format!("{:?}", e));
-        }
-    } else if config.rtt.enabled {
-        let session = Arc::new(Mutex::new(session));
+        let gdb_connection_string = config.gdb.gdb_connection_string.clone();
+        let session = session.clone();
+        std::thread::spawn(move || {
+            let gdb_connection_string = gdb_connection_string
+                .as_deref()
+                .or_else(|| Some("localhost:1337"));
+            // This next unwrap will always resolve as the connection string is always Some(T).
+            log::info!(
```

Comment:
What's the reason for this change? Is there a threading issue with the `logging::println` call?

Reply:
Yes, this writes into the RTTUI, messing with the entire layout, rendering RTT unusable :/ I have not fond a way to swallow the logs from the GDB thread yet :/
Currently the GDB server also emits a lot of warnings that maybe should be debug logs. We sould clean this up.